File: src/optic/lua/ssa/instructions/Declare.java
package optic.lua.ssa.instructions;

import optic.lua.ssa.*;

public final class Declare implements Step {
	private final String name;

	public Declare(String name) {
		this.name = name;
	}

	@Override
	public String toString() {
		return "local " + name;
	}
}

File: src/optic/lua/asm/instructions/Declare.kt
package optic.lua.asm.instructions

import optic.lua.asm.Step
import optic.lua.asm.StepVisitor
import optic.lua.asm.VariableInfo

class Declare(val variable: VariableInfo) : Step {
    override fun <T : Any, X : Throwable> accept(visitor: StepVisitor<T, X>): T = visitor.visitDeclaration(variable)

    override fun toString(): String {
        val type = variable.mode.toString().toLowerCase()
        return "local-$type $variable"
    }
}
